diesel/src/mysql/types/date_and_time.rs

```diff
@@ -78,16 +78,17 @@ impl ToSql<Timestamp, Mysql> for NaiveDateTime {
 impl FromSql<Timestamp, Mysql> for NaiveDateTime {
     fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<Error+Send+Sync>> {
         let mysql_time = <ffi::MYSQL_TIME as FromSql<Timestamp, Mysql>>::from_sql(bytes)?;
-        Ok(NaiveDate::from_ymd(
+
+        NaiveDate::from_ymd_opt(
             mysql_time.year as i32,
             mysql_time.month as u32,
             mysql_time.day as u32,
-        ).and_hms_micro(
+        ).and_then(|v| v.and_hms_micro_opt(
             mysql_time.hour as u32,
             mysql_time.minute as u32,
             mysql_time.second as u32,
             mysql_time.second_part as u32,
-        ))
+        )).ok_or_else(|| "Invalid date.".into())
```

Comment:
As this is a cold path (I hope!), can you make the error message a bit more elaborate to make this easier to debug? Something like `.ok_or_else(|| format!("Cannot parse this date: {:?}", mysql_time).into())`

Reply:
Sure thing! I copied the error message from over [here](https://github.com/diesel-rs/diesel/blob/3ae353c3aff8a7ea64ed6cb39b3a045ac86cd60e/diesel/src/sqlite/types/date_and_time/chrono.rs#L25-L28). I am guessing it would be a good idea to add something like this everywhere `from_ymd` is used.